tests/unit/s2n_tls13_zero_length_payload_test.c

```diff
@@ -43,7 +43,13 @@ int main(int argc, char **argv)
 
     EXPECT_SUCCESS(s2n_enable_tls13_in_test());
 
-    /* Test 0 length application data record handled gracefully in client and server mode */
+    /** Test 0 length application data record handled gracefully in client and server mode
+     * 
+     *= https://tools.ietf.org/rfc/rfc8446#section-5.4
+     *= type=test
+     *# Application Data records may contain a zero-length 
+     *# TLSInnerPlaintext.content if the sender desires.
```

Comment:
Is it possible to write an integration test for this case? It's an interesting one that we may want to validate our assumptions for.

Reply:
It should be possible. I can see that sending zero bytes over yields success for both s2nc/s2nd. The only issue is that I'm seeing unexpected record sizes from openssl. This might be how openssl deals with zero-sized payloads; I'm thinking we can investigate this after this pr.

Reply:
What do you mean by "sending zero bytes over yields success for both s2nc/s2nd"?

And unexpected record sizes from openssl makes me worry that some assumption in this PR is wrong. I don't think there's any rush on this test, so let's at least verify that you aren't seeing a real bug.

Reply:
I meant that when we we sent `data_bytes(0)`, the handshake completes successfully. What I'm unsure about is that I see openssl sending records of the form:

```
write to 0x107ede0 [0x109a7f3] (24 bytes => 24 (0x18))
0000 - 17 03 03 00 13
```

The 0x13 = 19 bytes is unexpected. I would have expected just 16 bytes from the tag or maybe 17 bytes if we sent a new line or return carriage character. This is unexpected for me but this is only inconsistent in the zero byte case, which leads be to think there is special behaviour in this case.